aten/src/ATen/native/xnnpack/Allocator.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <ATen/native/xnnpack/Common.h>
+
+#ifdef USE_XNNPACK
+
+namespace at {
+namespace native {
+namespace xnnpack {
+namespace internal {
+
+// XNNPACK can out-of-bound access XNN_EXTRA_BYTES beyond the end of the input
+// and output tensors.  This behavior will trigger ASAN, and can result in a
+// segfault if the accessed memory just so happens to fall on a different page
+// the current process has no read access to.
+// Here we define a custom allocator that allocates the extra storage required
+// to keep this behavior safe.
+
+class Allocator final : public at::Allocator {
```

Comment:
looks like we already have this hack implemented in aten/src/ATen/quantized/Quantizer.cpp - can we share the code?

Reply:
Do you have any preference as to where I should put it?

Reply:
I put it under native/utils for now.  If you have any other suggestions I am all ears.